src/mbgl/programs/segment.hpp

```diff
@@ -0,0 +1,40 @@
+#pragma once
```

Comment:
I think it's better here. The division as I see it is:

* `programs` -- code closely related to OpenGL, but specific to Mapbox GL in some way (so doesn't belong in `gl`). `segment.hpp` is similar to `attributes.hpp` and `uniforms.hpp` in this sense.
* `renderer` -- code that works with the Mapbox GL domain model (sources, layers, etc.) to facilitate rendering.